protocol-designer/src/organisms/SelectWellsModal/index.tsx

```diff
@@ -69,11 +69,20 @@ export const SelectWellsModal = (
     ? arrayToWellGroup(wellFieldData as string[])
     : {}
 
-  const [
-    selectedPrimaryWells,
-    setSelectedPrimaryWells,
-  ] = React.useState<WellGroup>(initialSelectedPrimaryWells)
-  const [highlightedWells, setHighlightedWells] = React.useState<WellGroup>({})
+  const [selectedPrimaryWells, setSelectedPrimaryWells] = useState<WellGroup>(
+    initialSelectedPrimaryWells
+  )
+  const [highlightedWells, setHighlightedWells] = useState<WellGroup>({})
+
+  useEffect(() => {
+    if (Array.isArray(wellFieldData)) {
+      if (wellFieldData.length === 0) {
+        setSelectedPrimaryWells({})
+      } else {
+        setSelectedPrimaryWells(arrayToWellGroup(wellFieldData as string[]))
+      }
+    }
+  }, [wellFieldData])
```

Comment:
maybe this is a bit cleaner

```suggestion
  const [selectedPrimaryWells, setSelectedPrimaryWells] = useState<WellGroup>(
    Array.isArray(wellFieldData)
      ? arrayToWellGroup(wellFieldData as string[])
      : {}
  )

  useEffect(() => {
    if (Array.isArray(wellFieldData)) {
      setSelectedPrimaryWells(
        wellFieldData.length === 0
          ? {}
          : arrayToWellGroup(wellFieldData as string[])
      )
    }
  }, [wellFieldData])
```